src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/RuntimeBasicWriter.cs

```diff
@@ -147,7 +147,7 @@ public override void WriteHtmlAttribute(CSharpRenderingContext context, HtmlAttr
         {
             var valuePieceCount = node
                 .Children
-                .Count(child => child is HtmlAttributeValueIRNode || child is CSharpAttributeValueIRNode);
+                .Count(child => child is HtmlAttributeValueIRNode || child is CSharpExpressionAttributeValueIRNode);
```

Comment:
Hmm? I'm just saying that the `valuePieceCount` will be wrong if you're not also checking if the child is a statement.